config/src/config.rs

```diff
@@ -71,17 +73,36 @@ impl Config {
     pub fn parse(prefix: ConfigPrefix, filename: &PathBuf) -> Result<Self, String> {
         let config_defaults = include_str!("../default_values.toml");
 
+        let mut config_content = std::fs::read_to_string(filename)
+            .map_err(|e| format!("Failed to read config file: {}", e))?;
+
+        config_content = Self::substitute_env_vars(config_content)?;
+
         let config: ConfigWrapper = Figment::new()
             .merge(Toml::string(config_defaults))
-            .merge(Toml::file(filename))
+            .merge(Toml::string(&config_content))
             .merge(Env::prefixed(prefix.get_prefix()).split("__"))
             .extract()
             .map_err(|e| e.to_string())?;
-        config.0.validate()?;
 
+        config.0.validate()?;
         Ok(config.0)
     }
 
+    fn substitute_env_vars(content: String) -> Result<String, String> {
```

Comment:
This can be just "String" instead of Result

Reply:
Sure, sorry for that